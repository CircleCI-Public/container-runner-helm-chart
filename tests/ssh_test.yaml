suite: test ssh
tests:
  - it: should create Gateway and Service resources if enabled
    template: templates/ssh.yaml
    set:
      agent.ssh.enabled: true
      agent.ssh.controllerName: "example.net/gateway-controller"
      agent.ssh.parametersRef:
        group: "test.group"
        kind: "testKind"
        name: "testName"
        namespace: "testNamespace"
      agent.ssh.startPort: 22
      agent.ssh.numPorts: 3
    asserts:
      - matchSnapshot: {}

  - it: should update Role rules to allow get Gateway and Service resources if enabled
    template: templates/role.yaml
    set:
      agent.ssh.enabled: true
    asserts:
      - contains:
          path: rules
          content:
            apiGroups: ["", "gateway.networking.k8s.io"]
            resources: ["gateways", "services"]
            verbs: ["get"]
        documentIndex: 0

  - it: should set container-agent ssh configuration if enabled
    template: templates/deployment.yaml
    set:
      agent.ssh.enabled: true
    asserts:
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: KUBE_SSH_IS_ENABLED
            value: "true"
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: KUBE_SSH_GATEWAY_NAME
            value: "RELEASE-NAME-container-agent-ssh"
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: KUBE_SSH_SERVICE_NAME
            value: "RELEASE-NAME-container-agent-ssh"

  - it: should not create any Gateway or Service resources if disabled
    template: templates/ssh.yaml
    set:
      agent.ssh.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should not update Role rules to allow get Gateway and Service resources if disabled
    template: templates/role.yaml
    set:
      agent.ssh.enabled: false
    asserts:
      - notContains:
          path: rules
          content:
            apiGroups: ["", "gateway.networking.k8s.io"]
            resources: ["gateways", "services"]
            verbs: ["get"]
        documentIndex: 0

  - it: should not set container-agent ssh configuration if disabled
    template: templates/deployment.yaml
    set:
      agent.ssh.enabled: false
    asserts:
      - notContains:
          path: spec.template.spec.containers[0].env
          content:
            name: KUBE_SSH_IS_ENABLED
            value: "true"
      - notContains:
          path: spec.template.spec.containers[0].env
          content:
            name: KUBE_SSH_GATEWAY_NAME
            value: "RELEASE-NAME-container-agent-ssh"
      - notContains:
          path: spec.template.spec.containers[0].env
          content:
            name: KUBE_SSH_SERVICE_NAME
            value: "RELEASE-NAME-container-agent-ssh"
