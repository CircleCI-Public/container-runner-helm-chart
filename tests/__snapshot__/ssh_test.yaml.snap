should create Gateway and Service resources if enabled:
  1: |
    apiVersion: gateway.networking.k8s.io/v1
    kind: GatewayClass
    metadata:
      finalizers:
        - gateway-exists-finalizer.gateway.networking.k8s.io
      name: RELEASE-NAME-container-agent-ssh
      namespace: NAMESPACE
    spec:
      controllerName: example.net/gateway-controller
      parametersRef:
        group: test.group
        kind: testKind
        name: testName
        namespace: testNamespace
  2: |
    apiVersion: gateway.networking.k8s.io/v1
    kind: Gateway
    metadata:
      name: RELEASE-NAME-container-agent-ssh
      namespace: NAMESPACE
    spec:
      gatewayClassName: RELEASE-NAME-container-agent-ssh
      listeners:
        - allowedRoutes:
            kinds:
              - kind: TCPRoute
          name: ssh-22
          port: 22
          protocol: TCP
        - allowedRoutes:
            kinds:
              - kind: TCPRoute
          name: ssh-23
          port: 23
          protocol: TCP
        - allowedRoutes:
            kinds:
              - kind: TCPRoute
          name: ssh-24
          port: 24
          protocol: TCP
  3: |
    apiVersion: gateway.networking.k8s.io/v1alpha2
    kind: TCPRoute
    metadata:
      name: ssh-22
      namespace: NAMESPACE
    spec:
      parentRefs:
        - name: RELEASE-NAME-container-agent-ssh
          sectionName: ssh-22
      rules:
        - backendRefs:
            - name: RELEASE-NAME-container-agent-ssh
              port: 22
  4: |
    apiVersion: gateway.networking.k8s.io/v1alpha2
    kind: TCPRoute
    metadata:
      name: ssh-23
      namespace: NAMESPACE
    spec:
      parentRefs:
        - name: RELEASE-NAME-container-agent-ssh
          sectionName: ssh-23
      rules:
        - backendRefs:
            - name: RELEASE-NAME-container-agent-ssh
              port: 23
  5: |
    apiVersion: gateway.networking.k8s.io/v1alpha2
    kind: TCPRoute
    metadata:
      name: ssh-24
      namespace: NAMESPACE
    spec:
      parentRefs:
        - name: RELEASE-NAME-container-agent-ssh
          sectionName: ssh-24
      rules:
        - backendRefs:
            - name: RELEASE-NAME-container-agent-ssh
              port: 24
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-container-agent-ssh
      namespace: NAMESPACE
    spec:
      ports:
        - name: ssh-22
          port: 22
          protocol: TCP
          targetPort: ssh-22
        - name: ssh-23
          port: 23
          protocol: TCP
          targetPort: ssh-23
        - name: ssh-24
          port: 24
          protocol: TCP
          targetPort: ssh-24
      selector:
        app.kubernetes.io/managed-by: circleci-container-agent
